src/music/midifile.cpp

```diff
@@ -149,8 +139,8 @@ class ByteBuffer {
 	bool ReadBuffer(byte *dest, size_t length)
 	{
 		if (this->IsEnd()) return false;
-		if (this->buflen - this->pos < length) return false;
-		memcpy(dest, this->buf + this->pos, length);
+		if (this->buf.size() - this->pos < length) return false;
+		std::copy(std::begin(this->buf) + this->pos, std::begin(this->buf) + this->pos + length, dest);
```

Comment:
Can't use `this->buf.begin()` ?

Reply:
My understanding is that std::begin() is preferred in modern C++.

Reply:
I see, it's more generic, and allows to easily change `this->buf` underlying type.